### PR TITLE
feat(sawtooth-ledger): add single sawtooth test ledger image

### PR DESCRIPTION
**Pull request by outSH**
feat(sawtooth-ledger): add single sawtooth test ledger image

- Enclose starting sawtooth ledger into single container that will fetch and run the entire setup.
- Remove reduntant files, update the readme.
- Add SawtoothTestLedger class to simplify setting the test ledger in jest tests.
- Refactor electricity-trade to generate usage without sawtooth shell patching.
- Refactor electricity-trade to use new sawtooth ledger container and wait until it's healthy.

Closes: https://github.com/hyperledger/cactus/issues/2108

Depends on: https://github.com/hyperledger/cactus/pull/2030

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

### CODE REVIEW

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Review by izuru0 — APPROVED** [review]
LGTM

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH 

1. FYI: The official image is now published here: https://github.com/orgs/hyperledger/packages/container/package/cactus-sawtooth-all-in-one
2. Could you please also add the publis CI yaml file in the .github/workflows dir next to the rest of the container publishing files? Easiest is to just copy paste one of the existing ones and then you mostly just tell it where the dockerfile is and how to build it (please be sure to use a docker build command in that yaml file that works from the project root directory without having to change directories inwards first.

File: packages/cactus-test-tooling/src/main/typescript/sawtooth/sawtooth-test-ledger.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  readonly useRunningLedger?: boolean;
+}
+
+/**
+ * Default options for Sawtooth test ledger.
+ */
+const DEFAULTS = Object.freeze({
+  // @todo Replace with hyperledger ghcr link when available
+  containerImageName: "ghcr.io/outsh/cactus-sawtooth-all-in-one",
+  containerImageVersion: "dev1",
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-test-tooling/src/main/typescript/sawtooth/sawtooth-test-ledger.ts]: Done

File: packages/cactus-test-tooling/src/main/typescript/sawtooth/sawtooth-test-ledger.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+  // For test development, attach to ledger that is already running, don't spin up new one
+  readonly useRunningLedger?: boolean;
+}
+
+/**
+ * Default options for Sawtooth test ledger.
+ */
+const DEFAULTS = Object.freeze({
+  // @todo Replace with hyperledger ghcr link when available
+  containerImageName: "ghcr.io/outsh/cactus-sawtooth-all-in-one",
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-test-tooling/src/main/typescript/sawtooth/sawtooth-test-ledger.ts]: Done

**Comment by outSH** [comment]
>     1. FYI: The official image is now published here: https://github.com/orgs/hyperledger/packages/container/package/cactus-sawtooth-all-in-one

Thanks!

>     2. Could you please also add the publis CI yaml file in the .github/workflows dir next to the rest of the container publishing files? Easiest is to just copy paste one of the existing ones and then you mostly just tell it where the dockerfile is and how to build it (please be sure to use a docker build command in that yaml file that works from the project root directory without having to change directories inwards first.

Oh, forgot about that, thanks for reminder - done :)

**Review by petermetz — APPROVED** [review]
> Oh, forgot about that, thanks for reminder - done :)
 
@outSH My pleasure, thank you for making the updates! LGTM!

**Pull request merged**